examples/navier-stokes/navierstokes.c

```diff
@@ -356,7 +356,9 @@ int main(int argc, char **argv) {
   const CeedInt dim = 3, ncompx = 3, ncompq = 5;
   CeedVector xcorners, xceed, qdata, q0ceed, mceed, onesvec, multlvec;
   CeedBasis basisx, basisxc, basisq;
-  CeedTransposeMode lmodeceed = CEED_NOTRANSPOSE, lmodepetsc = CEED_TRANSPOSE;
+
+  CeedLayoutMode lmodeceed = CEED_LAYOUT_COMPONENT_NODE,
+                 lmodepetsc = CEED_LAYOUT_NODE_COMPONENT;
```

Comment:
Thanks for doing this @jeremylt ! I'm not quite sure if this is the proper variable name. I know it's just a name, but petsc must not be the only package out there using this layout. Could we think of using something that is reminiscent of being "interlaced" or "consecutive/contiguous"?

Reply:
I am happy to change it if there is a more conventional name. My hesitation with 'interlaced' is that it does not specify *what* is interlaced. Same thing with 'contiguous' or 'consecutive'. Nodes? Fields?

Reply:
I use interlaced when explaining, and believe it's pretty unambiguous. This clip is from Gropp et al, 2001. Some people/projects use "blocked" as the converse, which I find highly ambiguous compared to "noninterlaced". In the hardware/vendor world, they tend to speak of SoA (structure of array) and AoS (array of structure) for this concept. (Vendors tend to recommend SoA for better vectorization, ignoring the cache locality consequences.)

![Screenshot_20200129_103140](https://user-images.githubusercontent.com/3303/73381198-8c955500-4282-11ea-9cce-245b8fc27325.png)

![Screenshot_20200129_103033](https://user-images.githubusercontent.com/3303/73381085-666fb500-4282-11ea-976d-0fd345d03607.png)

Reply:
`CEED_LAYOUT_NODE_INTERLACED` vs `CEED_LAYOUT_FIELD_INTERLACED`?

Reply:
or `CEED_LAYOUT_INTERLACED` and `CEED_LAYOUT_NONINTERLACED`. (I'm just making sure we have a decision so I only change this once more)

Reply:
I prefer interlaced and noninterlaced. I'd even consider `CeedInterlaceMode` and `CEED_INTERLACED` and `CEED_NONINTERLACED`. "Layout" is pretty ambiguous. The man page for `CeedInterlaceMode` can have a detailed description with examples.